### PR TITLE
[tp-6] custom modal component

### LINKED ISSUE #6: create custom popup component
create custom popup (you can use bootstrap)

### PR DESCRIPTION
**Pull request by altNameForStudying**
Modal not centered vertically: 
image

Modal centered vertically: 
image

### CODE REVIEW

File: src/components/shared/Modal/Modal.jsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    const handleClose = () => setShow(false);
+    console.log(props);
+    return (
+        <>
+            <Modal show={show} centered={props.isCentered}>
+                <Modal.Header closeButton onHide={handleClose}>
+                    <Modal.Title>{props.title}</Modal.Title>
+                </Modal.Header>
+
+                <Modal.Body>{<p>{props.body}</p>}</Modal.Body>
```
Comment (pdybowski) [inline src/components/shared/Modal/Modal.jsx]: props.children

end then when you init modal you do:
`<CustomModal>
    <SomeModal></SomeModal>
</CustomModal>`

**Review by pdybowski — CHANGES_REQUESTED** [review]

File: src/components/shared/Modal/Modal.jsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+                <Modal.Footer>
+                    <Button variant="secondary" onClick={handleClose}>
```
Comment (pdybowski) [inline src/components/shared/Modal/Modal.jsx]: default

File: src/components/shared/Modal/Modal.jsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { Button, Modal } from 'react-bootstrap';
+import { useState } from 'react';
+
+export const CustomModal = (props) => {
+    const [show, setShow] = useState(true);
+    const handleClose = () => setShow(false);
+    console.log(props);
+    return (
+        <>
```
Comment (pdybowski) [inline src/components/shared/Modal/Modal.jsx:22]: I think you dont need it

File: src/components/shared/Modal/Modal.jsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+                    <Modal.Title>{props.title}</Modal.Title>
+                </Modal.Header>
+
+                <Modal.Body>{<p>{props.body}</p>}</Modal.Body>
+
+                <Modal.Footer>
+                    <Button variant="secondary" onClick={handleClose}>
+                        {props.buttondismisstext}
+                    </Button>
+                    <Button variant="primary" onClick={handleClose}>
```
Comment (pdybowski) [inline src/components/shared/Modal/Modal.jsx]: handleSave

**Comment by pdybowski** [comment]
moreover add logic to change 'save' button to be also 'delete'

File: src/components/shared/Modal/Modal.jsx (hunk 1 above)
Comment (altNameForStudying) [reply src/components/shared/Modal/Modal.jsx]: Could you elaborate what you mean by 
`end then when you init modal you do:` ?

File: src/components/shared/Modal/Modal.jsx (hunk 2 above)
Comment (altNameForStudying) [reply src/components/shared/Modal/Modal.jsx]: Not sure what you mean by that. Do you mean passing props for variant selection?

**Pull request merged**